modules/service_accounts/data.tf

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+data "aws_iam_instance_profile" "tfe" {
+  count = var.existing_iam_instance_profile_name == null ? 0 : 1
+  name  = var.existing_iam_instance_profile_name
+}
+
+data "aws_iam_role" "instance_role" {
+  count = var.existing_iam_instance_role_name == null ? 0 : 1
+  name  = var.existing_iam_instance_role_name
+}
```

Comment:
Not a blocker, would help to modify these resources with the term "existing" so that later on this could read like: 

`iam_instance_role    = try(data.aws_iam_role.existing_instance_role[0], aws_iam_role.instance_role[0])`

Reply:
Ah, yeah, I like that. Thanks! I'll add it on Monday when I'm back from PTO. (Headed out now.)

Reply:
I had time after all, but that means I need another approval. :)